address.js

```diff
@@ -0,0 +1,263 @@
+const {validateBuffer, validateDerivationScheme, validateArray} = require("./utils/validation")
+const {chacha20poly1305Encrypt, chacha20poly1305Decrypt, blake2b, sha3_256} = require("./crypto-primitives")
+
+const cbor = require('borc')
+const crc32 = require('./utils/crc32')
+const base58 = require('./utils/base58')
+const CborIndefiniteLengthArray = require('./utils/CborIndefiniteLengthArray')
+
+function validateAddressType(input, addressTypeId) {
+    if (!['BASE', 'POINTER', 'ENTERPRISE', 'BOOTSTRAP'].includes(addressTypeId)) {
+        throw new Error("Invalid address type to validate!")
+    }
+    const addressType = {
+        BASE_ADDRESS_KEY_KEY: 0,
+        BASE_ADDRESS_SCRIPT_KEY: 1,
+        BASE_ADDRESS_KEY_SCRIPT: 2,
+        BASE_ADDRESS_SCRIPT_SCRIPT: 3,
+        POINTER_ADDRESS_KEY: 4,
+        POINTER_ADDRESS_SCRIPT: 5,
+        ENTERPRISE_ADDRESS_KEY: 6,
+        ENTERPRISE_ADDRESS_SCRIPT: 7,
+        BOOTSTRAP_ADDRESS: 8
+    }
+    const baseAddressTypes = [
```

Comment:
this can be shortened by declaring a separate map for each base, enterprise and pointer addresses and then using `...` operator to merge those keys in a single `addressTypes` map

see https://javascript.info/destructuring-assignment#the-rest-pattern